pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_metadata.go

```diff
@@ -58,6 +58,16 @@ func (r *metaDataReconciler) reconcile(ctx context.Context, workspace *tenancyv1
 		}
 	}
 
+	if workspace.Status.Phase == tenancyv1alpha1.ClusterWorkspacePhaseReady {
+		if _, found := workspace.Annotations[tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey]; found && len(workspace.Annotations) == 1 {
+			workspace.Annotations = nil
+			changed = true
+		} else if found {
+			delete(workspace.Annotations, tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey)
+			changed = true
+		}
```

Comment:
just a nit, not sure if this reads better:
```
		if _, found := workspace.Annotations[tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey]; found {
			delete(workspace.Annotations, tenancyv1alpha1.ClusterWorkspaceOwnerAnnotationKey)
			changed = true
		}

		if len(workspace.Annotations) == 0 {
			workspace.Annotations = nil
			changed = true
		}
```

Reply:
please gdoc why setting the annotation map to nil is necessary.

Reply:
it's not necessary actually. Removed.